bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/PowerControlPower.java

```diff
@@ -29,31 +33,82 @@
  */
 @NonNullByDefault
 public class PowerControlPower extends BaseChannelHandler<CommandConfirmation> {
+    private static final int WOL_PACKET_RETRY_COUNT = 10;
+    private static final int WOL_PACKET_RETRY_DELAY_MILLIS = 100;
+
     private final Logger logger = LoggerFactory.getLogger(PowerControlPower.class);
+    private final ConfigProvider configProvider;
+    private final ScheduledExecutorService scheduler;
+
+    public PowerControlPower(ConfigProvider configProvider, ScheduledExecutorService scheduler) {
+        this.configProvider = configProvider;
+        this.scheduler = scheduler;
+    }
 
     @Override
     public void onReceiveCommand(String channelId, LGWebOSHandler handler, Command command) {
+        final State state = handler.getSocket().getState();
         if (RefreshType.REFRESH == command) {
             handler.postUpdate(channelId, handler.getSocket().isConnected() ? OnOffType.ON : OnOffType.OFF);
-            return;
-        }
-        if (OnOffType.ON == command) {
-            logger.debug(
-                    "Received ON - Turning TV on via API is not supported by LG WebOS TVs. You may succeed using wake on lan binding, please consult lgwebos binding documentation.");
-        }
-        if (!handler.getSocket().isConnected()) {
-            /*
-             * Unable to send anything to a not connected device.
-             * onDeviceReady nor onDeviceRemoved will be called and item state would be permanently inconsistent.
-             * Therefore setting state to OFF
-             */
-            handler.postUpdate(channelId, OnOffType.OFF);
+        } else if (OnOffType.ON == command) {
+            if (State.DISCONNECTED == state) {
```

Comment:
Please move this to the switch statement

Reply:
will do